### ISSUE TITLE
Add search for the KB

### ISSUE THREAD

**Issue by itaysk** [aquasecurity/kube-hunter#240]
Currently the search is more like a lookup, where you can lookup an article by it's ID. We need a full text search to allow people to search for content freely.

**Comment by iYehuda** [comment]
Can we run some backend logic on GitHub pages to do that?
What is the common way to do this on that platform?

**Comment by itaysk** [comment]
No. We can do client side, or server side hosted as a service. There are several of those that work with Jekyll like algolia/lunr just to give an example

**Comment by RobbiNespu** [comment]
how about something like this
![Screenshot-20191024102538-962x673](https://user-images.githubusercontent.com/3077875/67448126-a5b9b480-f648-11e9-8c4e-ca1275895daf.png)

**Comment by itaysk** [comment]
looks good, but should replace or integrate with the current search in the sidebar. 
better to discuss the behavior and design of the feature here, and the implementation in the PR, so I'm adding [WIP] to your PR just so maintainers don't review it yet. Please remove it once you are ready to review.

**Comment by itaysk** [comment]
I think it would be best to use some external search indexing service. One option is Google custom search. It's free but ad supported. Another is Algolia which is free for not-commercial project which I guess we qualify.

**Comment by RaitoBezarius** [comment]
That sounds overkill, this seems a more appropriate solution if possible: http://elasticlunr.com/

**Comment by itaysk** [comment]
@RaitoBezarius if that can support full text indexing (of the content) that's an option too
